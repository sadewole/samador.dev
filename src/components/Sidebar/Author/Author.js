import React from "react"
import { withPrefix, Link } from "gatsby"

const Author = ({ author }) => {
  return (
    <div className="mb-4">
      <Link to="/">
        <img
          src={withPrefix(author.photo)}
          className="rounded"
          width="75"
          height="75"
          alt={author.name}
        />
      </Link>
      (
      <h2 className="text-3xl mb-2 leading-normal font-semibold text-white">
        <Link className="" to="/">
          {author.name}
        </Link>
      </h2>
      <p className="text-lg font-light leading-relaxed mt-0 mb-4 text-gray-300">
        {author.bio}
      </p>
    </div>
  )
}

export default Author
