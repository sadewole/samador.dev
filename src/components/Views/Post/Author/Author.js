import React from "react"
import PropTypes from "prop-types"
import { withPrefix } from "gatsby"
import { getContact } from "../../../../utils"

const Author = ({ author }) => {
  return (
    <div className="my-5 flex items-center">
      <img
        src={withPrefix(author.photo)}
        width="60"
        height="60"
        className="rounded mr-3"
        alt={author.name}
      />
      <div>
        <p>
          Written by{" "}
          <a
            className=""
            href={getContact("twitter", author.contacts.twitter)}
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>{author.name}</strong>
          </a>{" "}
        </p>
        <p className="text-sm font-light leading-relaxed mt-0 mb-4 text-gray-300">
          {author.bio}.
        </p>
      </div>
    </div>
  )
}

Author.propTypes = {
  author: PropTypes.object,
}

export default Author
