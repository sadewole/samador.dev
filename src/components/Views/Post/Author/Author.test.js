import React from "react"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import Author from "./Author"

describe("Author", () => {
  const props = {
    author: {
      name: "test name",
      photo: "/photo.jpg",
      bio: "test",
      contacts: { twitter: "test twitter" },
    },
  }

  it("must be string", () => {
    const { getByText } = render(<Author {...props} />)
    getByText("Written by")
  })

  it("render correctly", () => {
    const tree = renderer.create(<Author {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
