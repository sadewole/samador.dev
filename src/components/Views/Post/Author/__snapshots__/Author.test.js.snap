// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author render correctly 1`] = `
<div
  className="my-5 flex items-center"
>
  <img
    alt="test name"
    className="rounded mr-3"
    height="60"
    src="/photo.jpg"
    width="60"
  />
  <div>
    <p>
      Written by
       
      <a
        className=""
        href="https://www.twitter.com/test twitter"
        rel="noopener noreferrer"
        target="_blank"
      >
        <strong>
          test name
        </strong>
      </a>
       
    </p>
    <p
      className="text-sm font-light leading-relaxed mt-0 mb-4 text-gray-300"
    >
      test
      .
    </p>
  </div>
</div>
`;
